File: projects/blognami/lib/blognami/models/tag.rb

require "slick/database/row"

Slick::Database::Row.register "tag" do
  include :pageable

  has_many :tagable_tags
  has_many :tagables, through: [:tagable_tags, :tagable]

  table_class.define_method :to_field_value do
    out = []
    order_by(:name).each do |tag|
        out << tag.name
    end
    out.join("\n")
  end

end

File: packages/@blognami/tags/lib/models/tag.js

export default {
    meta(){
        this.include('pageable');
        this.include('sitemappable');

        this.hasMany('tagableTags');
        this.hasMany('tagables', { through: ['tagableTags', 'tagable'] });

        this.mustNotBeBlank('name');
    }
};
